test/switchboard/default-switchboards/FastSwitchboard.t.sol

```diff
@@ -5,104 +5,305 @@ import "../../Setup.t.sol";
     error WatcherFound();
     error WatcherNotFound();
     error AlreadyAttested();
     error InvalidSigLength();
-
+    error InvalidRoot();
+    error NoPermit(bytes32 role);
     FastSwitchboard fastSwitchboard;
 
     function setUp() external {
         initialise();
+        uint256[] memory transmitterPivateKeys = new uint256[](1);
+        transmitterPivateKeys[0] = _transmitterPrivateKey;
+
+        _dualChainSetup(transmitterPivateKeys);
+
+        packetId = bytes32(uint256(_a.chainSlug) << 224);
+
+        bytes32 digest = keccak256(
+            abi.encode(versionHash, _b.chainSlug, packetId, root)
+        );
+        bytes memory sig_ = _createSignature(digest, _transmitterPrivateKey);
+        _proposeOnDst(_b, sig_, packetId, root);
 
-        _a.chainSlug = uint32(c++);
-        _a.sigVerifier__ = new SignatureVerifier(_socketOwner);
-        remoteChainSlug = uint32(c++);
-        packetId = bytes32(uint256(remoteChainSlug) << 224);
+        assertEq(_b.socket__.packetIdRoots(packetId, 0), root);
 
         vm.startPrank(_socketOwner);
 
+        // fastSwitchboard = FastSwitchboard(
+        //     address(_b.configs__[0].switchboard__)
+        // );
+
         fastSwitchboard = new FastSwitchboard(
             _socketOwner,
-            address(uint160(c++)),
-            _a.chainSlug,
+            address(_b.socket__),
+            _b.chainSlug,
             1,
             _a.sigVerifier__
         );
 
         fastSwitchboard.grantRole(GOVERNANCE_ROLE, _socketOwner);
 
-        watcher = vm.addr(_watcherPrivateKey);
-        altWatcher = vm.addr(_altWatcherPrivateKey);
-
-        fastSwitchboard.grantWatcherRole(remoteChainSlug, watcher);
-        fastSwitchboard.grantWatcherRole(remoteChainSlug, altWatcher);
+        fastSwitchboard.grantWatcherRole(_a.chainSlug, _watcher);
+        fastSwitchboard.grantWatcherRole(_a.chainSlug, _altWatcher);
 
         vm.stopPrank();
     }
 
-    function testAttest() external {
+    function signAndPropose(
+        uint32 chainSlug_,
+        bytes32 packetId_,
+        bytes32 root_
+    ) internal {
         bytes32 digest = keccak256(
-            abi.encode(address(fastSwitchboard), _a.chainSlug, packetId)
+            abi.encode(versionHash, chainSlug_, packetId_, root_)
         );
-        bytes memory sig = _createSignature(digest, _watcherPrivateKey);
+        bytes memory sig_ = _createSignature(digest, _transmitterPrivateKey);
+        _proposeOnDst(_b, sig_, packetId, root_);
+    }
 
+    function testAttest() external {
         vm.expectEmit(false, false, false, true);
-        emit PacketAttested(packetId, watcher);
-
-        fastSwitchboard.attest(packetId, sig);
+        emit ProposalAttested(packetId, 0, root, _watcher, 1);
+        _attestOnDst(
+            address(fastSwitchboard),
+            _b.chainSlug,
+            packetId,
+            0,
+            _watcherPrivateKey
+        );
+        assertTrue(fastSwitchboard.isAttested(_watcher, root));
```

Comment:
Also check attestations[root]. Not checked in any of the test cases.
We know when this variable is updated and so you might feel isAttested check is enough. But the tests need to be dumb in this way to ensure we can catch errors in future changes.